libpromises/evalfunction.c

```diff
@@ -7777,6 +7780,190 @@ static int JsonPrimitiveComparator(JsonElement const *left_obj,
     return StringSafeCompare(left, right);
 }
 
+static bool ClassFilterDataArrayOfArrays(
+    EvalContext *ctx,
+    const char *fn_name,
+    JsonElement *json_array,
+    const char *class_expr_index,
+    bool *remove)
+{
+    errno = 0; /* to detect error */
+    assert(SIZE_MAX >= ULONG_MAX); /* make sure returned value can fit in size_t */
+    char *endptr;
+    size_t index = strtoul(class_expr_index, &endptr, 10);
+    if (!StringEqual(endptr, "")) /* check that the whole string was consumed */
+    {
+        Log(LOG_LEVEL_VERBOSE,
+            "Function %s(): Bad class expression index '%s': Not a valid integer",
+            fn_name, class_expr_index);
+        return false;
+    }
```

Comment:
I think we have some functions for this `StringToLong` and `StringToUnsignedLong` maybe, did you check if you can use them?